src/piksel/baseapp.cpp

```diff
@@ -444,22 +442,51 @@ void BaseApp::mainLoop(Graphics& g) {
 }
 
 void BaseApp::updateFramebufferSize(int framebufferWidth, int framebufferHeight) {
+    #if defined(__APPLE__) && !defined(__EMSCRIPTEN__)
+    // This is a bit of a hack, but GLFW does not seem to provide a way of querying the monior of the current context,
+    // so this is the only way I could come up with of dealing with a mix between a native retina display and an external non-retina one.
+    static int oldFramebufferWidth = width;
+    static int oldFramebufferHeight = height;
+    static bool retinaDisplay = false;
+    if (framebufferWidth == 2*oldFramebufferWidth and framebufferHeight == 2*oldFramebufferHeight) {
```

Comment:
i am not happy about this hack.
first problem is, it is not macOS specific, this can happen on every OS that supports content scaling.
second problem is, 2 times scaling is not the only available scaling.
i think it would be better to look for a more universal solution, like https://www.glfw.org/docs/latest/window_guide.html#window_scale
i did not try it myself but `glfwSetWindowContentScaleCallback` looks very promising.